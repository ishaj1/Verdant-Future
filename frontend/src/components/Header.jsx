import { Link, useNavigate } from "react-router-dom";
import useAuth from "../hooks/useAuth";
import useLogout from "../hooks/useLogout";

export default function Header() {
  const { auth } = useAuth();
  const navigate = useNavigate();
  const logout = useLogout();

  const signOut = async () => {
    await logout();
    navigate("/");
  };
  return (
    <header>
      <nav>
        {auth?.username && <Link to="/projects">Organizations Directory</Link>}
        {auth?.username && (
          <Link to={`/profile/${auth?.username}`}>Profile</Link>
        )}
        {auth?.username && <button onClick={signOut}>Logout</button>}
      </nav>
    </header>
  );
}
