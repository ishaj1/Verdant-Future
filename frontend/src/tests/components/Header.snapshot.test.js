import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../../components/LoginForm';

jest.mock('../../hooks/useAuth', () => ({
  __esModule: true,
  default: jest.fn(() => ({ auth: { username: 'testuser' } })),
}));

jest.mock('../../hooks/useLogout', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe('Header snapshots', () => {
  it('matches the snapshot when user is authenticated', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot when user is not authenticated', () => {
    jest.clearAllMocks();
    jest.mock('../../hooks/useAuth', () => ({
      __esModule: true,
      default: jest.fn(() => ({ auth: null })),
    }));
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

