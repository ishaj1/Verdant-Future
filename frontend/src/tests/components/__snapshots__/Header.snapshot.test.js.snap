// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot when user is authenticated 1`] = `
<div
  className="LoginWindow"
>
  <div
    className="LoginContent"
  >
    <div
      className="TypeSelector"
    >
      <button
        className="selected"
        id="project"
        onClick={[Function]}
      >
        Project
      </button>
      <button
        className=""
        id="company"
        onClick={[Function]}
      >
        Company
      </button>
    </div>
    <div
      className="mt-10 mb-5"
    >
      <h1
        className="text-center text-2xl font-bold leading-9 tracking-tight text-gray-900 mb-3"
      >
        Sign in to your account 
      </h1>
      <p
        className="text-center text-md text-gray-500"
      >
         Not a member? 
        <a
          className="text-customGreen-300"
          href="/register"
        >
          Sign up
        </a>
      </p>
    </div>
    <form
      className="LoginForm"
      id="loginform"
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="block text-md w-full rounded-md border-0 px-3 py-2 shadow-sm ring-1 ring-inset ring-customGreen-300 focus:ring-5 focus:ring-inset focus:ring-customGreen-600 focus:outline-1 focus:outline-customGreen-400 focus:shadow-md sm:text-sm sm:leading-6"
        id="username"
        name="username"
        required={true}
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="block text-md w-full rounded-md border-0 px-3 py-2 shadow-sm ring-1 ring-inset ring-customGreen-300 focus:ring-5 focus:ring-inset focus:ring-customGreen-600 focus:outline-1 focus:outline-customGreen-400 focus:shadow-md sm:text-sm sm:leading-6"
        id="password"
        name="password"
        required={true}
        type="password"
      />
      <input
        className="flex w-full justify-center rounded-md bg-customGreen-500 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-customGreen-200 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;

exports[`Header component matches the snapshot when user is not authenticated 1`] = `
<div
  className="LoginWindow"
>
  <div
    className="LoginContent"
  >
    <div
      className="TypeSelector"
    >
      <button
        className="selected"
        id="project"
        onClick={[Function]}
      >
        Project
      </button>
      <button
        className=""
        id="company"
        onClick={[Function]}
      >
        Company
      </button>
    </div>
    <div
      className="mt-10 mb-5"
    >
      <h1
        className="text-center text-2xl font-bold leading-9 tracking-tight text-gray-900 mb-3"
      >
        Sign in to your account 
      </h1>
      <p
        className="text-center text-md text-gray-500"
      >
         Not a member? 
        <a
          className="text-customGreen-300"
          href="/register"
        >
          Sign up
        </a>
      </p>
    </div>
    <form
      className="LoginForm"
      id="loginform"
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="block text-md w-full rounded-md border-0 px-3 py-2 shadow-sm ring-1 ring-inset ring-customGreen-300 focus:ring-5 focus:ring-inset focus:ring-customGreen-600 focus:outline-1 focus:outline-customGreen-400 focus:shadow-md sm:text-sm sm:leading-6"
        id="username"
        name="username"
        required={true}
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="block text-md w-full rounded-md border-0 px-3 py-2 shadow-sm ring-1 ring-inset ring-customGreen-300 focus:ring-5 focus:ring-inset focus:ring-customGreen-600 focus:outline-1 focus:outline-customGreen-400 focus:shadow-md sm:text-sm sm:leading-6"
        id="password"
        name="password"
        required={true}
        type="password"
      />
      <input
        className="flex w-full justify-center rounded-md bg-customGreen-500 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-customGreen-200 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
        type="submit"
        value="Login"
      />
    </form>
  </div>
</div>
`;
